tests/integration/test_core.py

```diff
@@ -89,9 +91,64 @@ async def test_mysql_relation(
     await ops_test.model.deploy("charmed-osm-mariadb-k8s", application_name="mariadb")
     await ops_test.model.add_relation("wordpress", "mariadb:mysql")
     await ops_test.model.wait_for_idle()
+
     assert (
             await get_app_status("wordpress") == ops.model.ActiveStatus.name
     ), (
         "application status should be active once correct database connection info "
         "being provided via relation"
     )
+
+
+@pytest.mark.asyncio
+async def test_get_initial_password_action(default_admin_password):
+    """
+    arrange: after WordPress charm has been deployed
+    act: run get-initial-password action
+    assert: get-initial-password action should return default admin password
+    """
+    assert (
+            len(default_admin_password) > 8
+    ), "get-initial-password action should return the default password"
+
+
+@pytest.mark.asyncio
+async def test_wordpress_functionality(
+        ops_test: pytest_operator.plugin.OpsTest,
+        application_name,
+        default_admin_password
+):
+    """
+    arrange: after WordPress charm has been deployed and db relation established
+    act: test WordPress basic functionality (login, post, comment)
+    assert: WordPress works normally as a blog site
+    """
+    units = (await ops_test.model.get_status()).applications[application_name].units.values()
+    for unit in units:
+        unit_ip = unit.address
+        wp_client = WordpressClient(
```

Comment:
Do you think this would be better for a fixture to make the test shorter and easier to understand/ read?

Reply:
There are some things about scaling a container based charm I am not sure yet. Will update on this later, thanks!

Reply:
Fixed, thanks!